### DISCUSSION TITLE
How to Set Up MSVC Toolchain for Targeting Windows XP (32-bit) on Windows 11?

### DISCUSSION THREAD

**Discussion by armed** [rust9x/rust#45] [Q&A] [answered]
Hi everyone,

Windows is not my primary development OS, so please bear with me. I’m running Windows 11 (64-bit) and need to set up a toolchain to use `rust9x` for targeting Windows XP (32-bit).

Is there a standalone version of all the required MSVC build tools available, perhaps as an archive or a self-contained package?

I’d appreciate any guidance or suggestions on how to achieve this setup.

I tried GUI based tool called vs_BuildTools.exe it downloads and installs bunch of stuff everywhere. What I want to achieve is just make simple folder with all the needed stuff like in sample project (e.g. `X:\rustprojs\rust9x\r9x-buildtools\vc71\Vc7\PlatformSDK\Lib`).

Thanks in advance!

**Comment by seritools** [comment, marked as answer]
> I tried GUI based tool called vs_BuildTools.exe it downloads and installs bunch of stuff everywhere.

yup, installing the XP-supporting toolset should work for this. The relevant files should then be:

- Windows 7 SDK files installed with it, in `C:\Program Files (x86)\Microsoft SDKs\Windows\v7.1A\Lib`.
![image](https://github.com/user-attachments/assets/c55402b7-6155-4442-8d51-a5f4194a7760)
- The toolset files for the v141 toolset, for me it's in `C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.16.27023\lib\x86\`
![image](https://github.com/user-attachments/assets/fa279d87-7f6b-49c3-9282-68b0ff1c34c3)


> e.g. X:\rustprojs\rust9x\r9x-buildtools\vc71\Vc7\PlatformSDK\Lib.

that's VS2003 files copied over from an actual installation I did in a WinXP VM before.

> Is there a standalone version of all the required MSVC build tools available, perhaps as an archive or a self-contained package?

nope, not that I know of

**Reply by armed** [reply]
Thanks, @seritools. That helped.

**Reply by Unknown78** [reply]
I couldn't find `C++ Windows XP Support for VS 2017 (v141) tools` with the currently latest `July 2026 Feature Update (18.8.0)` of `Visual Studio Build Tools 2026` of https://aka.ms/vs/stable/vs_BuildTools.exe

---

Okay, here's the news, https://developercommunity.visualstudio.com/t/11121716
> Nguyen Vinh (BEYONDSOFT CONSULTING INC) [MSFT]  
> 
> We have investigated this behavior with our engineering team, and we can confirm that the removal of the VC++ v140 (VS 2015) and Windows XP support components in Visual Studio 2026 version 18.8 is By Design.

**Reply by seritools** [reply]
you might need the VS2022 or earlier version, not sure if it's still included with VS2026.

**Reply by seritools** [reply]
@Unknown78 i verified today, it definitely is in VS2022. Make sure to select both v141 x86 _and_ the Windows XP support. Also the Windows XP support in 2017 is limited to dynamically linked MSVCRT. A statically linked one will link unsupported code.
